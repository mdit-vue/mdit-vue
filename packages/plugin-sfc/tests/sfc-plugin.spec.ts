import type { MarkdownItEnv } from '@mdit-vue/types';
import MarkdownIt from 'markdown-it';
import { expect, it } from 'vitest';
import { sfcPlugin } from '../src/index.js';

const source = `\
# hello vuepress

{{ msg }}

<docs>
extra hoisted tag
</docs>

<script setup lang="ts">
const foo = 'scriptSetup'
</script>

<script>
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
</script>

<style lang="stylus">
.h1
  red
</style>
`;

it('should extract default sfc blocks correctly', () => {
  const md = MarkdownIt({ html: true }).use(sfcPlugin);
  const env: MarkdownItEnv = {};

  const rendered = md.render(source, env);

  expect(rendered).toMatchSnapshot();
  expect(env.sfcBlocks).toMatchSnapshot();
});

it('should extract custom blocks correctly', () => {
  const md = MarkdownIt({ html: true }).use(sfcPlugin, {
    customBlocks: ['docs'],
  });
  const env: MarkdownItEnv = {};

  const rendered = md.render(source, env);

  expect(rendered).toMatchSnapshot();
  expect(env.sfcBlocks).toMatchSnapshot();
});
