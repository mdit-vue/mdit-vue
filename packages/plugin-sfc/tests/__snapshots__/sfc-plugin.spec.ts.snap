// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should extract custom blocks correctly 1`] = `
"<h1>hello vuepress</h1>
<p>{{ msg }}</p>
"
`;

exports[`should extract custom blocks correctly 2`] = `
{
  "customBlocks": [
    {
      "content": "<docs>
extra hoisted tag
</docs>",
      "contentStripped": "
extra hoisted tag
",
      "tagClose": "</docs>",
      "tagOpen": "<docs>",
      "type": "docs",
    },
  ],
  "script": {
    "content": "<script>
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
</script>",
    "contentStripped": "
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
",
    "tagClose": "</script>",
    "tagOpen": "<script>",
    "type": "script",
  },
  "scriptSetup": {
    "content": "<script setup lang="ts">
const foo = 'scriptSetup'
</script>",
    "contentStripped": "
const foo = 'scriptSetup'
",
    "tagClose": "</script>",
    "tagOpen": "<script setup lang="ts">",
    "type": "script",
  },
  "scripts": [
    {
      "content": "<script setup lang="ts">
const foo = 'scriptSetup'
</script>",
      "contentStripped": "
const foo = 'scriptSetup'
",
      "tagClose": "</script>",
      "tagOpen": "<script setup lang="ts">",
      "type": "script",
    },
    {
      "content": "<script>
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
</script>",
      "contentStripped": "
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
",
      "tagClose": "</script>",
      "tagOpen": "<script>",
      "type": "script",
    },
  ],
  "styles": [
    {
      "content": "<style lang="stylus">
.h1
  red
</style>",
      "contentStripped": "
.h1
  red
",
      "tagClose": "</style>",
      "tagOpen": "<style lang="stylus">",
      "type": "style",
    },
  ],
  "template": {
    "content": "<template><h1>hello vuepress</h1>
<p>{{ msg }}</p>
</template>",
    "contentStripped": "<h1>hello vuepress</h1>
<p>{{ msg }}</p>
",
    "tagClose": "</template>",
    "tagOpen": "<template>",
    "type": "template",
  },
}
`;

exports[`should extract default sfc blocks correctly 1`] = `
"<h1>hello vuepress</h1>
<p>{{ msg }}</p>
<docs>
extra hoisted tag
</docs>
"
`;

exports[`should extract default sfc blocks correctly 2`] = `
{
  "customBlocks": [],
  "script": {
    "content": "<script>
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
</script>",
    "contentStripped": "
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
",
    "tagClose": "</script>",
    "tagOpen": "<script>",
    "type": "script",
  },
  "scriptSetup": {
    "content": "<script setup lang="ts">
const foo = 'scriptSetup'
</script>",
    "contentStripped": "
const foo = 'scriptSetup'
",
    "tagClose": "</script>",
    "tagOpen": "<script setup lang="ts">",
    "type": "script",
  },
  "scripts": [
    {
      "content": "<script setup lang="ts">
const foo = 'scriptSetup'
</script>",
      "contentStripped": "
const foo = 'scriptSetup'
",
      "tagClose": "</script>",
      "tagOpen": "<script setup lang="ts">",
      "type": "script",
    },
    {
      "content": "<script>
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
</script>",
      "contentStripped": "
export default {
  setup() {
    return {
      msg: 'script'
    }
  }
}
",
      "tagClose": "</script>",
      "tagOpen": "<script>",
      "type": "script",
    },
  ],
  "styles": [
    {
      "content": "<style lang="stylus">
.h1
  red
</style>",
      "contentStripped": "
.h1
  red
",
      "tagClose": "</style>",
      "tagOpen": "<style lang="stylus">",
      "type": "style",
    },
  ],
  "template": {
    "content": "<template><h1>hello vuepress</h1>
<p>{{ msg }}</p>
<docs>
extra hoisted tag
</docs>
</template>",
    "contentStripped": "<h1>hello vuepress</h1>
<p>{{ msg }}</p>
<docs>
extra hoisted tag
</docs>
",
    "tagClose": "</template>",
    "tagOpen": "<template>",
    "type": "template",
  },
}
`;
